src/Factory.sol

```diff
@@ -0,0 +1,149 @@
+    /// @param _addressResolver: address of synthetix address resolver
+    /// @param _settings: address of settings for accounts
+    /// @param _ops: contract address for gelato ops -- must be payable
+    constructor(
+        address _owner,
+        bytes32 _version,
+        address _marginAsset,
+        address _addressResolver,
+        address _settings,
+        address payable _ops
+    ) {
+        /// @dev transfer ownership to owner
+        transferOwnership(_owner);
+
+        version = _version;
+        marginAsset = _marginAsset;
+        addressResolver = _addressResolver;
+        settings = _settings;
+        ops = _ops;
+
+        /// @dev deploy logic for proxy
+        logic = new Account();
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                           ACCOUNT DEPLOYMENT
+    //////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc IFactory
+    function newAccount()
+        external
+        override
+        returns (address payable accountAddress)
+    {
+        /// @dev ensure one account per address
+        if (creatorToAccount[msg.sender] != address(0)) {
+            revert AlreadyCreatedAccount(creatorToAccount[msg.sender]);
+        }
+
+        // create account
+        accountAddress = payable(
+            _cloneAsMinimalProxy(address(logic), "Creation failure")
+        );
+
+        // update creator to account mapping
+        creatorToAccount[msg.sender] = accountAddress;
+
+        // initialize new account
+        Account account = Account(accountAddress);
+        account.initialize(
+            address(marginAsset),
+            addressResolver,
+            settings,
+            ops
+        );
+
+        // transfer ownership of account to caller
+        account.transferOwnership(msg.sender);
+
+        emit NewAccount(msg.sender, accountAddress, version);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                                SETTERS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice
+    /// Factory state is mutable to allow for upgrades. The following setters
+    /// will *NOT* impact existing accounts. Only *NEW* accounts will be affected.
+    /// The setters are only callable by the owner of the factory.
+
+    /// @inheritdoc IFactory
+    function setVersion(bytes32 _version) external override onlyOwner {
+        version = _version;
+    }
+
+    /// @inheritdoc IFactory
+    function setLogic(address payable _logic) external override onlyOwner {
+        logic = Account(_logic);
+    }
+
+    /// @inheritdoc IFactory
+    function setSettings(address _settings) external override onlyOwner {
+        settings = _settings;
+    }
+
+    /// @inheritdoc IFactory
+    function setMarginAsset(address _marginAsset) external override onlyOwner {
+        marginAsset = _marginAsset;
+    }
```

Comment:
Can be dangerous if you set some and forget others. What is the issue with re-deploying the factory any time we upgrade?